src/Functions/FunctionsConversion.h

```diff
@@ -2509,6 +2509,7 @@ class ExecutableFunctionCast : public IExecutableFunction
     }
 
     bool useDefaultImplementationForNulls() const override { return false; }
+    bool useDefaultImplementationForNothing() const override { return false; }
```

Comment:
Ok, here it is reasonable cause the return type for CAST(Nothing, 'T') should be T, not Nothing

Reply:
Added comment